mbpfan.spec

```diff
@@ -0,0 +1,68 @@
+Name:           mbpfan
+URL:            https://github.com/dgraziotin/mbpfan
+License:        GPLv3
+Group:          System Environment/Daemons
+Version:        %{SOURCE_VERSION}
+Release:        1
+Summary:        A simple daemon to control fan speed on all MacBook/MacBook Pros (probably all Apple computers) for Linux 3.x.x and 4.x.x
+Source:         v%{version}.tar.gz
+
+%description
+This is an enhanced version of Allan McRae mbpfan
+
+mbpfan is a daemon that uses input from coretemp module and sets the fan speed using the applesmc module. This enhanced version assumes any number of processors and fans (max. 10).
+
+It only uses the temperatures from the processors as input.
+It requires coretemp and applesmc kernel modules to be loaded.
+It requires root use
+It daemonizes or stays in foreground
+Verbose mode for both syslog and stdout
+Users can configure it using the file /etc/mbpfan.conf
+
+%prep
+%setup -q -n %{name}-%{version}
+
+%build
+make
+
+%install
+install -D -m755 bin/mbpfan $RPM_BUILD_ROOT/usr/sbin/mbpfan
+install -D -m644 mbpfan.conf $RPM_BUILD_ROOT/etc/mbpfan.conf
+install -D -m644 mbpfan.service $RPM_BUILD_ROOT/usr/lib/systemd/system/mbpfan.service
+
+%clean
+rm -rf $RPM_BUILD_ROOT
+
+%post
+%systemd_post mbpfan.service
+# If it is a first installation then autoconfigure daemon
+if [[ ${1} == 1 ]]; then
+  min_fan_speed="$(cat /sys/devices/platform/applesmc.768/fan*_min| uniq| sort |head -1)"
+  max_fan_speed="$(cat /sys/devices/platform/applesmc.768/fan*_max| uniq| sort |tail -1)"
+  let max_temp=$(cat /sys/devices/platform/coretemp.*/hwmon/hwmon*/temp*_max|uniq |sort |tail -1)/1000
```

Comment:
I wonder if we should add similar autodetect logic as above?  On my system these are:

```
$ cat /sys/devices/platform/coretemp.0/temp*max
100000
100000
100000
```

However `mbpfan.c` contains the comment:

```c
int max_temp = 86;   // do not set it > 90
```

Perhaps we should cap it at 90, or maybe the comment is incorrect?

Reply:
You're right. I missed that. Probably we could cap it at 90 (as the author suggest).

Reply:
Again, should we put this logic in mbpfan?

Reply:
Definitely! Maybe we should make mbpfan guess sane settings for the machine is running on and let the user override this values by the config file. At that point the config file is superfluous and the user will benefit of an (extra) simple installation.